### ISSUE TITLE
Debug spins after hitting break point and trying to continue 

### ISSUE THREAD

**Issue by LawG4** [devkitPro/libogc#131]
Hello,

I've been trying to get debugging working through the debug library inside libogc. 
The program is running on Dolphin Emulator v 5.0-16793, and it is emulating usb gecko over port 55020 in slot A. I'm using gdb v13.0.50.20220721 and the latest commit on libogc v[3d123eb](https://github.com/devkitPro/libogc/commit/3d123ebd2babebea0e59330f085fd40edb8c7233)

```c
DEBUG_Init(GDBSTUB_DEVICE_USB, 1);
_break();
```

I set a break point with _break(), the program is executed and hits that break point. I then open the elf's symbols in gdb and connect to dolphin via 

> target remote <Dolphin's Local IP Address>:55020

The debugger attaches to the process, and I can print the stack frame, and see the line of code that the debugger is breaking on. This is all fantastic, but I then try to step or continue and the process hangs.

I've been pausing the emulator at random points to evaluate the call stack and also adding debug strings to see which parts are hit - but I'm a bit out of my depth at the moment, I wanted to check if this is the correct approach for gdb debugging with devkitPro? Or should I using a different method?

Further details that may be of use if this is an actual bug and not user error on my behalf:
 The loop seems to go:
* debug_handler.S
* call c_debug_handler
* within c_debug_handler - resolve it's a command and return
* return back to debug_handler.S
* return to the start of _break 
* Rerun _break
* debug_handler.S

Interestingly the using gdb return command will return the program counter to the line one above return address, but when attempting to step or continue after a return, you get taken to the breakpoint immediately instead. 

All the best,
Lawrence

**Comment by WinterMute** [comment]
Right now the latest compatible gdb client is the 7.7.1 we supply, sorry. You can obtain a binary via (sudo) (dkp-)pacman -S devkitPPC-gdb. 

Using a bleeding edge gdb is not recommended. There have been quite a lot of changes which aren't always that well documented and often don't take account of the older hardware we're attempting to support.

**Comment by LawG4** [comment]
Hi, thanks for the reply!
So I went back to using the version in devkitpro package `/opt/devkitpro/devkitPPC/bin/powerpc-eabi-gdb` and I found some problems. 

The powerpc-eabi-gcc in devkitpro will generate debug symbols with dwarf version 5 by default. But the gdb 7.7.1 will only read up to dwarf4. Adding `-gdwarf-4` to the CC flags will build my code with dwarf4, this includes rebuilding libogc with dwarf 4 as well.

As without rebuilding libogc, because it is a static library the dwarf 5 symbols get embedded into the elf - which means the
